aten/src/ATen/core/ivalue.cpp

```diff
@@ -953,9 +953,31 @@ std::vector<std::reference_wrapper<const at::DataPtr>> ivalue::Future::extractDa
   if (value.isPyObject()) {
     std::vector<at::Tensor> tensors =
         value.toPyObjectHolder()->extractTensors();
-    data_ptrs.reserve(tensors.size());
+    auto reserve_size = 0;
     for (const at::Tensor& tensor : tensors) {
-      data_ptrs.emplace_back(tensor.storage().data_ptr());
+      if (tensor.is_sparse()) {
+        // Sparse tensor is indices and values. Both are tensors
+        // and contain storage. Therefore reserve_size needs to be
+        // incremented by 2.
+        reserve_size += 2;
+      } else {
+        // A dense/strided tensor contains 1 storage.
+        reserve_size += 1;
+      }
+    }
```

Comment:
This is much nicer, thanks! The only suggestion I can make is that comments in code are good, but self-explanatory code is even better! (Comments go "stale" quickly, because people forget to update them, whereas this cannot happen to code). Here, for example, if you called the variable `num_storages` then the code itself would already convey most of the meaning that is currently in the comment.

In general I try to only use comments when they're unavoidable, and never to explain _what_ the code is doing: comments should only provide context to explain _why_ some code is doing things in a way that would otherwise look unreasonable.

Reply:
I don't think `num_storages` is a good name because the code is storing storage data_ptr in data_ptrs.

Reply:
okay Ill update it.